__tests__/types.test.js

```diff
@@ -0,0 +1,106 @@
+const {
+  isNumber,
+  isString,
+  isBoolean,
+  isArray,
+  isObject,
+  isFunction,
+  castToNumber,
+  castToString,
+  castToBoolean,
+  getCaster
+} = require('../lib/types.js');
+  
+describe('validator module', () => {
+  describe('basic validation', () => {
+    it('properly tells if a value is a numbers', () => {
+      expect(isNumber(3)).toBeTruthy();
+      expect(isNumber('hi')).toBeFalsy();
+      expect(isNumber([])).toBeFalsy();
+      expect(isNumber({})).toBeFalsy();
+      expect(isNumber(() => {})).toBeFalsy();
+      expect(isNumber(true)).toBeFalsy();
+    });
+    it('properly tells if a value is a string', () => {
+      expect(isString('hi')).toBeTruthy();
+      expect(isString('true')).toBeTruthy();
+      expect(isString('')).toBeTruthy();
+      expect(isString('2')).toBeTruthy();
+      expect(isString(2)).toBeFalsy();
+      expect(isString([])).toBeFalsy();
+      expect(isString({})).toBeFalsy();
+      expect(isString(true)).toBeFalsy();
+    });
+    it('properly tells if a value is a boolean', () => {
+      expect(isBoolean(true)).toBeTruthy();
+      expect(isBoolean(false)).toBeTruthy();
+      expect(isBoolean('true')).toBeFalsy();
+      expect(isBoolean('false')).toBeFalsy();
+      expect(isBoolean(2)).toBeFalsy();
+      expect(isBoolean(0)).toBeFalsy();
+      expect(isBoolean([])).toBeFalsy();
+    });
+    it('properly tells if a value is an array', () => {
+      expect(isArray([])).toBeTruthy();
+      expect(isArray(['cat'])).toBeTruthy();
+      expect(isArray({ 2: 2 })).toBeFalsy();
+      expect(isArray(2)).toBeFalsy();
+      expect(isArray('what')).toBeFalsy();
+      expect(isArray(false)).toBeFalsy();
+    });
+    it('properly tells if a value is an object', () => {
+      expect(isObject([])).toBeTruthy();
```

Comment:
ideally you'd want this to be false. in our validation schemas we wouldn't want arrays to be counted as objects. you'd have to do some more logic in your isObject function, but its doable.